shared/src/telemetry/data.rs

```diff
@@ -0,0 +1,55 @@
+use serde_json::json; // Ensure you have `serde_json` in your Cargo.toml
+use serde_json::Value;
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+// Function to get the PostgreSQL data directory from the PGDATA environment variable
+pub fn get_postgres_data_directory() -> Option<String> {
+    env::var("PGDATA").ok()
```

Comment:
Where is this environment variable coming from? Why can't we use the `pg_sys::GetConfigOptionByName` to get this directory?

Reply:
Every Postgres distro sets PGDATA as the environment variable for the Postgres data directory. Unfortunately, we can't use `pg_sys` because it requires the library to be a Postgres extension and have a `.control` file, which we don't do for `shared/`.

This is robust. I've tested it on Bitnami, Postgres, and `pgrx`